modelexpress_client/python/tests/test_refit_generator_client.py

```
@@ -0,0 +1,303 @@
+        "rank:0",
+        "rank:1",
+    ]
+
+
+def test_generator_releases_lease_when_manifest_is_invalid(monkeypatch):
+    server, endpoint, service = _start_server(manifest_digest="bad-digest")
+    adapter = _Adapter(service)
+    generator = _initialize(monkeypatch, endpoint, adapter)
+
+    try:
+        with pytest.raises(RuntimeError, match="no usable source.*digest mismatch"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)